code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ZeroBezelNotificationBuilderTest.kt

```diff
@@ -0,0 +1,234 @@
+        mockBundle = MockAEPPushTemplateDataProvider.getMockedAEPBundleWithAllKeys()
+        every { getCachedImage(any()) } answers { mockBitmap }
+    }
+
+    @Test
+    fun `verify construct with image downloaded and collapsedStyle is img`() {
+        val pushTemplate = ZeroBezelPushTemplate(MapData(dataMap))
+        every { cacheImages(any()) } answers { 2 }
+        every { anyConstructed<RemoteViews>().setImageViewBitmap(any(), mockBitmap) } just Runs
+
+        val notification = ZeroBezelNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass
+        ).build()
+
+        assertEquals(pushTemplate.channelId, notification.channelId)
+        verifyNotificationDataFields(notification, pushTemplate)
+        verify { cacheImages(listOf(pushTemplate.imageUrl)) }
+        verify { getCachedImage(pushTemplate.imageUrl) }
+        verify { anyConstructed<RemoteViews>().setImageViewBitmap(any(), mockBitmap) }
+    }
+
+    @Test
+    fun `verify construct with image downloaded and collapsedStyle is txt`() {
+        dataMap.replaceValueInMap(
+            Pair(
+                PushTemplateConstants.PushPayloadKeys.ZERO_BEZEL_COLLAPSED_STYLE,
+                "txt"
+            )
+        )
+        val pushTemplate = ZeroBezelPushTemplate(MapData(dataMap))
+
+        every { cacheImages(any()) } answers { 2 }
+        every { anyConstructed<RemoteViews>().setImageViewBitmap(any(), mockBitmap) } just Runs
+        every { anyConstructed<RemoteViews>().setViewVisibility(any(), View.GONE) } just Runs
+
+        val notification = ZeroBezelNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass
+        ).build()
+
+        assertEquals(pushTemplate.channelId, notification.channelId)
+        verifyNotificationDataFields(notification, pushTemplate)
+        verify { cacheImages(listOf(pushTemplate.imageUrl)) }
+        verify { getCachedImage(pushTemplate.imageUrl) }
+        verify { anyConstructed<RemoteViews>().setImageViewBitmap(any(), mockBitmap) }
+        verify { anyConstructed<RemoteViews>().setViewVisibility(any(), View.GONE) }
+    }
+
+    @Test
+    fun `verify construct with no image downloaded`() {
+        val pushTemplate = ZeroBezelPushTemplate(MapData(dataMap))
+
+        every { cacheImages(listOf(pushTemplate.imageUrl)) } returns 0
+        every { anyConstructed<RemoteViews>().setViewVisibility(any(), View.GONE) } just Runs
+
+        val notification = ZeroBezelNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass
+        ).build()
+
+        assertEquals(pushTemplate.channelId, notification.channelId)
+        verifyNotificationDataFields(notification, pushTemplate)
+        verify { cacheImages(listOf(pushTemplate.imageUrl)) }
+        verify { anyConstructed<RemoteViews>().setViewVisibility(any(), View.GONE) }
```

Comment:
can we do verify(exactly = 2) here since visibility of both expanded and collapsed layouts are GONE